runtime/sema/resolve.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2022 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sema
+
+import (
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/common"
+)
+
+type AddressContractNamesResolver func(address common.Address) ([]string, error)
+
+// AddressLocationHandlerFunc returns a location handler
+// which returns a single location for non-address locations,
+// and uses the given address contract names resolve function
+// to get all contract names for an address
+//
+func AddressLocationHandlerFunc(resolveAddressContractNames AddressContractNamesResolver) LocationHandlerFunc {
```

Comment:
This was extracted from the language server. Even though not used by the `runtime` package, it is still the "default" behaviour. A future PR refactor the LS to use this handler, too